flink-connectors/flink-connector-hbase/src/main/java/org/apache/flink/connector/hbase/source/HBaseSource.java

```diff
@@ -0,0 +1,135 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.hbase.source;
+
+import org.apache.flink.api.connector.source.Boundedness;
+import org.apache.flink.api.connector.source.Source;
+import org.apache.flink.api.connector.source.SourceReader;
+import org.apache.flink.api.connector.source.SourceReaderContext;
+import org.apache.flink.api.connector.source.SplitEnumerator;
+import org.apache.flink.api.connector.source.SplitEnumeratorContext;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.connector.hbase.source.enumerator.HBaseSourceEnumeratorCheckpointSerializer;
+import org.apache.flink.connector.hbase.source.enumerator.HBaseSplitEnumerator;
+import org.apache.flink.connector.hbase.source.reader.HBaseSourceDeserializer;
+import org.apache.flink.connector.hbase.source.reader.HBaseSourceReader;
+import org.apache.flink.connector.hbase.source.split.HBaseSourceSplit;
+import org.apache.flink.connector.hbase.source.split.HBaseSourceSplitSerializer;
+import org.apache.flink.connector.hbase.util.HBaseConfigurationUtil;
+import org.apache.flink.core.io.SimpleVersionedSerializer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+
+/**
+ * A source connector for HBase. Please use an {@link HBaseSourceBuilder} to construct an {@link
+ * HBaseSource}. The following example shows how to create a HBaseSource that reads String values
+ * from each cell.
+ *
+ * <pre>{@code
+ * HBaseSource<String> source =
+ *     HBaseSource.builder()
+ *         .setSourceDeserializer(new HBaseStringDeserializer())
+ *         .setTableName("test-table")
+ *         .setHBaseConfiguration(new HBaseTestClusterUtil().getConfig())
+ *         .build();
+ *
+ * static class HBaseStringDeserializer implements HBaseSourceDeserializer<String> {
+ *     @Override
+ *     public String deserialize(HBaseSourceEvent event) {
+ *         return new String(event.getPayload(), HBaseEvent.DEFAULT_CHARSET);
+ *     }
+ * }
+ * }</pre>
+ *
+ * @see HBaseSourceBuilder HBaseSourceBuilder for more details.
+ */
+public class HBaseSource<T> implements Source<T, HBaseSourceSplit, Collection<HBaseSourceSplit>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HBaseSource.class);
+
+    private static final long serialVersionUID = 1L;
+
+    private final HBaseSourceDeserializer<T> sourceDeserializer;
+    private final byte[] serializedHBaseConfig;
+    private final Configuration sourceConfiguration;
+
+    HBaseSource(
+            HBaseSourceDeserializer<T> sourceDeserializer,
+            org.apache.hadoop.conf.Configuration hbaseConfiguration,
+            Configuration sourceConfiguration) {
+        this.serializedHBaseConfig =
+                HBaseConfigurationUtil.serializeConfiguration(hbaseConfiguration);
+        this.sourceDeserializer = sourceDeserializer;
+        this.sourceConfiguration = sourceConfiguration;
+        LOG.debug("constructed source");
```

Comment:
Add config, you could also override `#toString` and add `this` here (or even better in `builder#build`).